File: frontend/src/types/demand.ts
/** 需求（悬赏求助） */
export interface Demand {
  id: number
  userId: number
  skillTagId: number | null
  title: string
  description: string
  pointReward: number
  deadline: string | null
  locationType: 1 | 2 | 3 // 1-线上 2-线下 3-均可
  longitude: number
  latitude: number
  campus: string
  building: string
  status: 0 | 1 | 2 | 3 // 0-已关闭 1-待接单 2-进行中 3-已完成
  createTime: string
  updateTime: string
  // 前端附加字段
  userName?: string
  userAvatar?: string
  tagName?: string
  categoryName?: string
  distance?: number
  isUrgent?: boolean // 是否急单（12h内截止）
}

/** 需求发布/编辑表单 */
export interface DemandForm {
  skillTagId: number | null
  title: string
  description: string
  pointReward: number
  deadline: string | null
  locationType: 1 | 2 | 3
  isUrgent: boolean
}

File: backend/dabashou-demand/src/main/java/com/dabashou/demand/domain/Demand.java
package com.dabashou.demand.domain;

import com.baomidou.mybatisplus.annotation.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@TableName("dbs_demand")
public class Demand {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long skillTagId;
    private String title;
    private String description;
    private Integer pointReward;
    private LocalDateTime deadline;
    private Integer locationType;
    private BigDecimal longitude;
    private BigDecimal latitude;
    private String campus;
    private String building;
    private Integer demandType;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public Long getSkillTagId() { return skillTagId; }
    public void setSkillTagId(Long skillTagId) { this.skillTagId = skillTagId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public Integer getPointReward() { return pointReward; }
    public void setPointReward(Integer pointReward) { this.pointReward = pointReward; }
    public LocalDateTime getDeadline() { return deadline; }
    public void setDeadline(LocalDateTime deadline) { this.deadline = deadline; }
    public Integer getLocationType() { return locationType; }
    public void setLocationType(Integer locationType) { this.locationType = locationType; }
    public BigDecimal getLongitude() { return longitude; }
    public void setLongitude(BigDecimal longitude) { this.longitude = longitude; }
    public BigDecimal getLatitude() { return latitude; }
    public void setLatitude(BigDecimal latitude) { this.latitude = latitude; }
    public String getCampus() { return campus; }
    public void setCampus(String campus) { this.campus = campus; }
    public String getBuilding() { return building; }
    public void setBuilding(String building) { this.building = building; }
    public Integer getDemandType() { return demandType; }
    public void setDemandType(Integer demandType) { this.demandType = demandType; }
    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }
    public LocalDateTime getCreateTime() { return createTime; }
    public void setCreateTime(LocalDateTime createTime) { this.createTime = createTime; }
    public LocalDateTime getUpdateTime() { return updateTime; }
    public void setUpdateTime(LocalDateTime updateTime) { this.updateTime = updateTime; }
}
